homeassistant/components/plugwise/gateway.py

```diff
@@ -64,95 +66,99 @@ async def async_setup_entry_gw(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    if entry.unique_id is None:
+        if api.smile_version[0] != "1.8.0":
+            hass.config_entries.async_update_entry(entry, unique_id=api.smile_hostname)
+
     update_interval = timedelta(
         seconds=entry.options.get(
             CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL[api.smile_type]
         )
     )
 
-    async def async_update_data():
+    async def async_update_gw_data():
         """Update data via API endpoint."""
+        _LOGGER.debug("Updating %s", api.smile_name)
         try:
-            async with async_timeout.timeout(DEFAULT_TIMEOUT):
-                await api.full_update_device()
+            async with async_timeout.timeout(update_interval.seconds):
+                await api.update_gw_devices()
                 return True
         except XMLDataMissingError as err:
             raise UpdateFailed("Smile update failed") from err
+        except PlugwiseException as err:
+            _LOGGER.debug("Updating failed, generic failure for %s", api.smile_name)
+            raise UpdateFailed("Smile update failed") from err
 
     coordinator = DataUpdateCoordinator(
         hass,
         _LOGGER,
-        name=f"Smile {api.smile_name}",
-        update_method=async_update_data,
+        name=f"{api.smile_name}",
+        update_method=async_update_gw_data,
         update_interval=update_interval,
     )
 
     await coordinator.async_config_entry_first_refresh()
 
-    api.get_all_devices()
-
-    if entry.unique_id is None and api.smile_version[0] != "1.8.0":
-        hass.config_entries.async_update_entry(entry, unique_id=api.smile_hostname)
-
     undo_listener = entry.add_update_listener(_update_listener)
 
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
-        "api": api,
+        API: api,
         COORDINATOR: coordinator,
         PW_TYPE: GATEWAY,
         UNDO_UPDATE_LISTENER: undo_listener,
     }
 
-    device_registry = await dr.async_get_registry(hass)
-    device_registry.async_get_or_create(
-        config_entry_id=entry.entry_id,
-        identifiers={(DOMAIN, api.gateway_id)},
-        manufacturer="Plugwise",
-        name=entry.title,
-        model=f"Smile {api.smile_name}",
-        sw_version=api.smile_version[0],
-    )
-
-    single_master_thermostat = api.single_master_thermostat()
+    api.get_all_devices()
 
-    platforms = PLATFORMS_GATEWAY
-    if single_master_thermostat is None:
+    platforms = GATEWAY_PLATFORMS
+    if api.single_master_thermostat() is None:
         platforms = SENSOR_PLATFORMS
 
-    hass.config_entries.async_setup_platforms(entry, platforms)
+    async def delete_notification(self):
+        """Service: delete the Plugwise Notification."""
+        try:
+            deleted = await api.delete_notification()
+            _LOGGER.debug("PW Notification deleted: %s", deleted)
+        except PlugwiseException:
+            _LOGGER.debug(
+                "Failed to delete the Plugwise Notification for %s", api.smile_name
+            )
```

Comment:
This is a new service, lets keep that out of this PR to keep PRs a minimal a possible.